go/libkb/kbsig.go

```diff
@@ -329,6 +359,27 @@ func (arg ProofMetadata) ToJSON(m MetaContext) (ret *jsonw.Wrapper, err error) {
 	ret.SetKey("seqno", jsonw.NewInt64(int64(seqno)))
 	ret.SetKey("prev", prev)
 
+	var hPrevInfo HPrevInfo
+	if (arg.Me == nil) == (arg.HPrevInfoFallback == nil) {
+		return nil, fmt.Errorf("Exactly one of arg.Me and arg.HPrevInfoFallback must be non-nil.")
+	} else if arg.Me != nil {
+		hPrevInfo, err = arg.Me.GetExpectedNextHPrevInfo()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		hPrevInfo = *arg.HPrevInfoFallback
+	}
+
+	hPrevInfoObj := jsonw.NewDictionary()
+	hPrevInfoObj.SetKey("seqno", jsonw.NewInt64(int64(hPrevInfo.Seqno)))
+	if hash := hPrevInfo.Hash; hash != nil {
+		hPrevInfoObj.SetKey("hash", jsonw.NewString(hash.String()))
+	} else {
+		hPrevInfoObj.SetKey("hash", jsonw.NewNil())
+	}
+	ret.SetKey("high_skip", hPrevInfoObj)
```

Comment:
Would it be better to skip this section if `hPrevInfo` is default-valued? I think it would be default-valued in the case of a team loaded from the previous version of the cache.

Reply:
Also what about putting hprev writes behind a feature flag?